hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -1953,6 +1954,24 @@ private void addToCompactingFiles(Collection<HStoreFile> filesToAdd) {
     Collections.sort(filesCompacting, storeEngine.getStoreFileManager().getStoreFileComparator());
   }
 
+  /**
+   * Remove the files from compacting files. This usually happens when we clear compaction queues.
+   */
+  void removeFromCompactingFiles(Collection<HStoreFile> filesToRemove) {
+    synchronized (filesCompacting) {
+      filesCompacting.removeAll(filesToRemove);
+      Collections.sort(filesCompacting, storeEngine.getStoreFileManager().getStoreFileComparator());
+    }
```

Comment:
Why need to sort here？

Reply:
I think we should always keep it sorted. some methods, like org.apache.hadoop.hbase.regionserver.compactions.SortedCompactionPolicy#getCurrentEligibleFiles(), may did not work if it is unsorted.

Reply:
Do you want to resolve this comment? Looks like there is no sorting anymore.